server/controllers/stock/index.js

```diff
@@ -129,78 +130,205 @@ function createStock(req, res, next) {
+    transaction.addQuery(`INSERT INTO lot SET ?`, {
+      uuid : db.bid(lotUuid),
+      label : lot.label,
+      initial_quantity : lot.quantity,
+      quantity : lot.quantity,
+      unit_cost : lot.unit_cost,
+      expiration_date : new Date(lot.expiration_date),
+      inventory_uuid : db.bid(lot.inventory_uuid),
+      origin_uuid : db.bid(identifier),
+      delay : 0,
+    });
+
+    // adding a movement insertion query into the transaction
+    transaction.addQuery(`INSERT INTO stock_movement SET ?`, {
+      uuid : db.bid(uuid()),
+      lot_uuid : db.bid(lotUuid),
+      depot_uuid : db.bid(params.movement.depot_uuid),
+      document_uuid : db.bid(documentUuid),
+      flux_id : params.movement.flux_id,
+      date : new Date(params.movement.date),
+      quantity : lot.quantity,
+      unit_cost : lot.unit_cost,
+      is_exit : 0,
+      user_id : params.movement.user_id,
+      description : params.movement.description,
+      period_id : periodId,
+    });
+  });
 
-        // adding a movement insertion query into the transaction
-        transaction.addQuery(`INSERT INTO stock_movement SET ?`, {
-          uuid : db.bid(uuid()),
-          lot_uuid : db.bid(lotUuid),
-          depot_uuid : db.bid(params.movement.depot_uuid),
-          document_uuid : db.bid(documentUuid),
-          flux_id : params.movement.flux_id,
-          date : new Date(params.movement.date),
-          quantity : lot.quantity,
-          unit_cost : lot.unit_cost,
-          is_exit : 0,
-          user_id : params.movement.user_id,
-          description : params.movement.description,
-          period_id : periodId,
-        });
-      });
+  const postingParams = [
+    db.bid(documentUuid), 0, session.project.id, session.enterprise.currency_id,
+  ];
 
-      const postingParams = [db.bid(documentUuid), 0, req.session.project.id, req.session.enterprise.currency_id];
+  if (session.enterprise.settings.enable_auto_stock_accounting) {
+    transaction.addQuery('CALL PostStockMovement(?)', [postingParams]);
+  }
 
-      if (req.session.enterprise.settings.enable_auto_stock_accounting) {
-        transaction.addQuery('CALL PostStockMovement(?)', [postingParams]);
-      }
+  await transaction.execute();
+  return documentUuid;
+}
 
-      // execute all operations as one transaction
-      return transaction.execute();
-    })
-    .then(() => {
+/**
+ * POST /stock/integration
+ * create a new integration entry
+ */
+function createIntegration(req, res, next) {
+  insertNewStock(req.session, req.body, 'integration')
+    .then(documentUuid => {
       res.status(201).json({ uuid : documentUuid });
     })
-    .catch(next)
-    .done();
+    .catch(next);
+}
+
+/**
+ * POST /stock/inventory_adjustment
+ * stock inventory adjustement
+ */
+async function createInventoryAdjustment(req, res, next) {
+  try {
+    const movement = req.body;
+
+    const lots = movement.lots
+      .filter(l => l.quantity !== l.oldQuantity);
```

Comment:
You do this check on the client, right?  Can we just skip this?

Reply:
Yeah i did that on the client, but i was also afraid if somehow these data were not filtered on the client